### ISSUE TITLE
Fix S3928 FP: Diagnostic is reported in local function for parameters of containing method 

### ISSUE THREAD

**Issue by hankovich** [SonarSource/sonar-dotnet#7631]
### Repro steps

```c#
public sealed class Class
{
    public void Foo(int i)
    {
        Local();

        void Local()
        {
            throw new ArgumentException("Hello", nameof(i)); // <- S3928 is reported here
        }
    }
}
```

### Related information

* C#/VB.NET Plugins version 9.5.0.73987
* MSBuild / dotnet version .NET 8 Preview 6

**Comment by sebastien-marichal** [comment]
Hello @hankovich,

Thank you for your feedback.

This is considered a true positive by our analyzer.

As for lambda (see #7630), local function execution is postponed (especially in asynchronous execution, see [S4457](https://sonarsource.github.io/rspec/#/rspec/S4457/csharp) for a concrete example), and so are any checks inside.
Since the parameter is captured, the check can be done before the local function definition and execution, adopting a fail-fast approach.

Hoping this helps!
